src/main/ts/components/Chart.tsx

```diff
@@ -49,25 +53,53 @@ export default function Chart() {
     caycStartingDate,
     chartStartingDate,
     chartEndDate,
+    setSelectedProjects,
   } = useData();
 
-  if (isLoading) {
-    return <Spinner />;
-  }
+  const handleLoadProjects = async (q: string, resolve: (options: ProjectOption[]) => void) => {
+    const components = await getProjects({ q });
+    if (!components || components.length === 0) {
+      resolve([]);
+      return [];
+    }
+    const options = [
+      defaultProject,
+      ...components.map(({ key, name }) => {
+        return {
+          value: key,
+          label: name,
+        } as ProjectOption;
+      }),
+    ];
+    resolve(options);
+    return options;
+  };
+
+  const debouncedLoadProjects = React.useRef(
+    debounce((q: string, resolve: (options: ProjectOption[]) => void) => {
+      handleLoadProjects(q, resolve);
+    }, DEBOUNCE_DELAY),
+  );
```

Comment:
The above comment should simplify this as you don't need the 2nd parameter anymore (`resolve`).

Reply:
Yes. The debounce might even work again.